tests/php/Http/Controllers/Api/VolumeControllerTest.php

```diff
@@ -211,4 +224,379 @@ public function testUpdateRedirect()
         $response->assertRedirect('/');
         $response->assertSessionHas('saved', true);
     }
+
+    public function testCloneVolumeApi()
+    {
+        $volume = $this->volume([
+            'created_at' => '2022-11-09 14:37:00',
+            'updated_at' => '2022-11-09 14:37:00',
+        ])->fresh(); // Use fresh() to load even the null fields.
+        // The target project.
+        $project = ProjectTest::create();
+
+        $this->doTestApiRoute('POST', "/api/v1/volumes/{$volume->id}/clone-to/{$project->id}");
+
+        $this->be($project->creator);
+        $this->postJson("/api/v1/volumes/{$volume->id}/clone-to/{$project->id}")
+            // No update permissions in the source project.
+            ->assertStatus(403);
+
+        $this->beAdmin();
+        $this->postJson("/api/v1/volumes/{$volume->id}/clone-to/{$project->id}")
+            // No update permissions in the target project.
+            ->assertStatus(403);
+
+        $project->addUserId($this->admin()->id, Role::adminId());
+        Cache::flush();
+        $this->postJson("/api/v1/volumes/{$volume->id}/clone-to/{$project->id}")
+            ->assertStatus(200);
+
+    }
+
+    public function testCloneVolume()
```

Comment:
You could actually merge this with `testCloneVolumeApi()` as this test only adds the assertions for the new volume.